cpp/cmake/thirdparty/get_gputreeshap.cmake

```diff
@@ -27,6 +27,41 @@ function(find_and_configure_gputreeshap)
             GIT_TAG         ${PKG_PINNED_TAG}
     )
 
+    if (GPUTreeShap_ADDED)
+        install(TARGETS GPUTreeShap
+                EXPORT  gputreeshap-exports)
+
+        # generate gputreeshap-targets.cmake for install dir
+        rapids_export(INSTALL GPUTreeShap
+            EXPORT_SET gputreeshap-exports
+            GLOBAL_TARGETS GPUTreeShap
+            NAMESPACE GPUTreeShap::)
+
+        # generate gputreeshap-targets.cmake for binary dir
+        rapids_export(BUILD GPUTreeShap
+            EXPORT_SET gputreeshap-exports
+            GLOBAL_TARGETS GPUTreeShap
+            NAMESPACE GPUTreeShap::)
+
+        # clear out incorrect INTERFACE_SOURCES
+        set_target_properties(GPUTreeShap PROPERTIES INTERFACE_SOURCES "")
```

Comment:
I believe you would want this cleared before the export, so that the export doesn't contain this header.